packages/mock/src/client.ts

```diff
@@ -229,15 +230,21 @@ round-trip min/avg/max/stddev = 10.977/14.975/23.159/4.790 ms
   }
 }
 
+export type MockRequestHandler = (method: HttpMethod, path: string, options: any) => Promise<any>;
 export class MockFetchClient {
   initialized = false;
   initPromise?: Promise<void>;
 
   constructor(
     readonly router: FhirRouter,
     readonly repo: MemoryRepository,
-    readonly debug = false
-  ) {}
+    readonly debug = false,
+    readonly mockHandlerOverride?: MockRequestHandler
```

Comment:
🫠 very mixed feelings about adding more complexity to `MockRepository`.... When it was just something that we used internally for testing, that was one thing.  As more people are using it in more places, I think we need to be a lot more thoughtful about API design here.

Reply:
Yeah I don't have strong opinions on: 
1. This as the API design
2. Allowing this kind of extension at all

The main thing I was solving for was how to mock out really large counts for the `SearchControl`, without actually creating a ton of resources. 

Open to suggestions